packages/@react-aria/autocomplete/src/useAutocomplete.ts

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps, BaseEvent, DOMProps, RefObject} from '@react-types/shared';
+import {AutocompleteProps, AutocompleteState} from '@react-stately/autocomplete';
+import {ChangeEvent, InputHTMLAttributes, KeyboardEvent as ReactKeyboardEvent, useCallback, useEffect, useMemo, useRef} from 'react';
+import {CLEAR_FOCUS_EVENT, FOCUS_EVENT, mergeProps, mergeRefs, UPDATE_ACTIVEDESCENDANT, useEffectEvent, useId, useLabels, useObjectRef} from '@react-aria/utils';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {useFilter, useLocalizedStringFormatter} from '@react-aria/i18n';
+import {useKeyboard} from '@react-aria/interactions';
+
+export interface CollectionOptions extends DOMProps, AriaLabelingProps {
+  /** Whether the collection items should use virtual focus instead of being focused directly. */
+  shouldUseVirtualFocus: boolean,
+  /** Whether typeahead is disabled. */
+  disallowTypeAhead: boolean
+}
+export interface AriaAutocompleteProps extends AutocompleteProps {
+  /**
+   * The filter function used to determine if a option should be included in the autocomplete list.
+   * @default contains
+   */
+  defaultFilter?: (textValue: string, inputValue: string) => boolean
```

Comment:
Wonder if this should just be called `filter`? Why did we call it `defaultFilter` in ComboBox? I guess it's not a public prop there, but only in the hook?